File: Data Structures & Algorithms/find-in-mountain-array/submission-1.py
class Solution:
    def findInMountainArray(self, target: int, mountainArr: 'MountainArray') -> int:
        length = mountainArr.length()

        low,high = 1,length-2

        while low <= high:
            mid = (low+high)//2
            left,center,right = mountainArr.get(mid-1),mountainArr.get(mid),mountainArr.get(mid+1)

            if left < center < right:
                low = mid+1
            elif left > center > right:
                high = mid-1
            else:
                break
        
        peak = mid

        low,high = 0,peak

        while low <=high:
            mid = (low+high)//2
            val = mountainArr.get(mid)

            if  val > target:
                high = mid-1
            elif val < target:
                low = mid+1
            else:
                return mid
        
        low,high = peak+1,length-1

        while low <= high:
            mid = (low+high)//2
            val = mountainArr.get(mid)
            if val < target:
                high = mid - 1
            elif val > target:
                low = mid+1
            else:
                return mid
        return -1

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {

        int left = 0;
        int right = numbers.length -1;
       int sum = 0;
       int [] ans = new int[2]; 
        System.out.println("left "+left+ " right "+right);
        while(left < right){
            sum = numbers[left] + numbers[right];
             System.out.println(sum);
             System.out.println("left "+left+ " right "+right);
            if(sum > target){
                right--;
            }
            if(sum<target){
                left++;
            }
            if(sum == target){
                break;
            }
            sum = 0;
        }
        ans[0]=left+1;
        ans[1]=right+1;
        return ans;
    }
}
